src/templates/dotnet/DotnetTemplateProvider.ts

```diff
@@ -62,24 +64,40 @@ export class DotnetTemplateProvider extends TemplateProviderBase {
         }
     }
 
-    public async getLatestTemplateVersion(): Promise<string> {
-        return await cliFeedUtils.getLatestVersion(this.version);
+    public async getLatestTemplateVersion(context: IActionContext): Promise<string> {
+        const projKey = await this.getProjKey();
+
+        let templateVersion = await cliFeedUtils.getLatestVersion(this.version);
+        let netRelease = await this.getNetRelease(projKey, templateVersion);
+        if (netRelease) {
+            return templateVersion;
+        } else {
+            for (const newVersion of Object.values(FuncVersion)) {
+                if (newVersion !== this.version) {
+                    templateVersion = await cliFeedUtils.getLatestVersion(newVersion);
+                    netRelease = await this.getNetRelease(projKey, templateVersion);
+                    if (netRelease) {
+                        context.telemetry.properties.effectiveProjectRuntime = newVersion;
+                        const oldMajorVersion = getMajorVersion(this.version);
+                        const newMajorVersion = getMajorVersion(newVersion);
+                        const warning = localize('mismatchProjKeyWarning', 'WARNING: "{0}" is not supported on Azure Functions v{1}. Using templates from v{2} instead.', projKey, oldMajorVersion, newMajorVersion);
+                        ext.outputChannel.appendLog(warning);
+                        return templateVersion;
+                    }
+                }
+            }
+            throw new Error(localize('projKeyError', '"{0}" is not supported for Azure Functions version "{1}".', projKey, this.version));
```

Comment:
Would it make sense to not throw an error and just let the debug happen?  They'll get an error whenever they start debugging, right?  I'm wondering if there are any configurations where it might work (like a -preview version or something) that the extension might not recognize

Reply:
This is for creating a function, not debugging